src/java.base/share/classes/sun/security/ssl/SSLKeyDerivation.java

```diff
@@ -30,6 +30,9 @@
 import javax.crypto.SecretKey;
 
 interface SSLKeyDerivation {
-    SecretKey deriveKey(String algorithm,
-            AlgorithmParameterSpec params) throws IOException;
+    SecretKey deriveKey(String purpose) throws IOException;
+
+    default byte[] deriveData(String purpose) throws IOException {
```

Comment:
This is an internal interface, so I don't think you need to make this a `default` method.

Reply:
I didn't add `deriveData(String)` impl to all the existing impls of `SSLKeyDerivation`. Only impls used for deriving IVs are updated to add impl for `deriveData(String)`, so the `default` method is necessary.